src/controllers/subscription.js

```diff
@@ -24,114 +30,178 @@ const getSubscriptionsByID = (req, res) => {
-          message: 'Subscription not found',
-          data: undefined,
-        });
-      }
-      return res.status(200).json({
-        error: false,
-        message: 'Subscription found',
-        data: subscription,
+
+  try {
+    const foundSubscriptionById = await Subscription.findById(id).populate('classes member');
+    if (!foundSubscriptionById) {
+      return res.status(404).json({
+        error: true,
+        message: 'Subscription not found',
+        data: undefined,
       });
-    })
-    .catch((error) => res.status(500).json({
+    }
+
+    return res.status(200).json({
+      error: false,
+      message: `Subscription found: ${foundSubscriptionById}`,
+      data: foundSubscriptionById,
+    });
+  } catch (error) {
+    return res.status(500).json({
       error: true,
-      message: error,
+      message: error.message,
       data: undefined,
-    }));
+    });
+  }
 };
 
-const createSubscription = (req, res) => {
+const createSubscription = async (req, res) => {
   const { classes, member, date } = req.body;
-  Subscription.create({
-    classes,
-    member,
-    date,
-  })
-    .then((newSubscription) => res.status(201).json({
+
+  if (!mongoose.isValidObjectId(classes) || !mongoose.isValidObjectId(member)) {
+    return res.status(400).json({
+      error: true,
+      message: 'Class or member ID is not valid',
+    });
+  }
+
+  try {
+    const existingClass = await Class.findById(classes);
+    const existingMember = await Member.findById(member);
+
+    if (!existingClass || !existingMember) {
+      return res.status(404).json({
+        error: true,
+        message: 'Class or member not found',
+      });
+    }
+
+    const existingSubscription = await Subscription.findOne({ classes, member, date });
+
+    if (existingSubscription) {
+      return res.status(400).json({
+        error: true,
+        message: 'Subscription already exists',
+        data: existingSubscription,
+      });
+    }
+
+    const newSubscription = await Subscription.create({
+      classes,
+      member,
+      date,
+    });
+
+    return res.status(201).json({
       error: false,
       message: 'Subscription created successfully',
       data: newSubscription,
-    }))
-    .catch((error) => res.status(500).json({
+    });
+  } catch (error) {
+    return res.status(500).json({
       error: true,
-      message: error,
+      message: error.message,
       data: undefined,
-    }));
+    });
+  }
 };
 
-const updateSubscription = (req, res) => {
+const updateSubscription = async (req, res) => {
   const { id } = req.params;
   const { classes, member, date } = req.body;
 
-  if (!mongoose.Types.ObjectId.isValid(id)) {
+  if (!mongoose.isValidObjectId(id)
+  || (classes && !mongoose.isValidObjectId(classes))
+  || (member && !mongoose.isValidObjectId(member))) {
     return res.status(400).json({
       error: true,
-      message: 'ID is not valid',
+      message: 'ID or class or member ID is not valid',
```

Comment:
I think the second message it's a little bit confusing with so many or, you could say something like "One of the IDs is invalid" or "Subscription, Class and Member IDs must have a valid ID"